### PR TITLE
Implement Gitalk as a comment provider

### PR DESCRIPTION
**Pull request by fletchto99**
I had some free time this week to take a first pass at implementing Gitalk as a comment provider for the theme. This PR implements Gitalk using jsdelivr as the CDN. It was heavily based off of the work done to implement gitment (thanks @deamwork).

Furthermore, this PR contains a few small typo fixes as well as updating the dependencies to their latest versions.

Gitalk does offer more [configuration options](https://github.com/gitalk/gitalk) which I would be happy to pass through as variable to the config, just let me know.

I should note that this won't be compatible with Gitment issues out of the box since I opted to use the title as the identifier (it's what I use locally since the URL was too long).

Demo: https://blog.fletchto99.com/

closes: https://github.com/LouisBarranqueiro/hexo-theme-tranquilpeak/issues/490

/cc @LouisBarranqueiro

---
This change is [Reviewable](https://reviewable.io/reviews/louisbarranqueiro/hexo-theme-tranquilpeak/533)

### CODE REVIEW

**Comment by fletchto99** [comment]
Looks like there might be some CI errors related to the `package.json` changes. I don't have the time to look into this right now. If you have an idea feel free to push up a fix. Alternatively I could revert back to the old package.json however it would be nice to get it up to date.

**Comment by fletchto99** [comment]
Friendly bump

/cc @LouisBarranqueiro

**Comment by LouisBarranqueiro** [comment]
👋 @fletchto99 thanks a lot for the contribution! You are gonna make some people happy 😄 

Could you please rebase your branch with the `dev` branch and re-push your changes.

> Furthermore, this PR contains a few small typo fixes as well as updating the dependencies to their latest versions.

I already took care of that since theme features and dev workflow need to be tested. Let's focus on the feature :)

> Gitalk does offer more configuration options which I would be happy to pass through as variable to the config, just let me know.

yes, let's do that so users can customize Gitalk the way they want it 😄 


> I should note that this won't be compatible with Gitment issues out of the box since I opted to use the title as the identifier (it's what I use locally since the URL was too long).

I'm not sure to understand. What do you mean exactly? Are you talking about your installation or in general if a user switch from Gitment to Gitalk?

File: _config.yml [hunk 1]
```diff
@@ -202,6 +202,20 @@ gitment:
...
+    # Your Github ID (Github username):
+    github_id:
+    # The repo to store comments:
+    repo:
+    # Your client ID:
+    client_id:
+    # Your client secret:
+    client_secret:
+    # Distraction Free:
+    distraction_free: false
```
Comment (LouisBarranqueiro) [inline _config.yml]: Regarding your suggestion:
> Gitalk does offer more configuration options which I would be happy to pass through as variable to the config, just let me know.

I suggest we keep `enable`, `github_id`, `repo`, `client_id`, `client_secret` at the root and then we add a key `options` so users can pass all other properties. E.g:
``` yaml
enable: true
owner: LouisBarranqueiro
repo: gitalk-blog-comments
client_id: 32f32fj230d23h23jf32
client_secret: jd2083mf8-4fn-3nf3f32
options:
    distractionFreeMode: false
    language: 'fr'
    flipMoveOptions:
        staggerDelayBy: 40
```
so we can easily spread and pass this options to Gitalk. What do you think?

File: _config.yml [hunk 2]
```diff
@@ -202,6 +202,20 @@ gitment:
     # Your client secret:
     client_secret:
 
+# Your Gitalk information
```
Comment (LouisBarranqueiro) [inline _config.yml]: How about adding `Read https://github.com/gitalk/gitalk#options to set up Gitalk`?

File: _config.yml [hunk 3]
```diff
@@ -202,6 +202,20 @@ gitment:
...
+gitalk:
+    # Switch
+    enable: false
+    # Your Github ID (Github username):
+    github_id:
```
Comment (LouisBarranqueiro) [inline _config.yml]: I suggest `owner` so we have the same wording as Gitalk and users can check Gitalk documentation to set it up. 

I also suggest we take the description of each option from the Gitalk documentation so it's super explicit.
``` yaml
# Your GitHub Application Client ID
client_id:
# Your GitHub Application Client Secret
client_secret:
# Your GitHub username
owner:
# The GitHub repository which will host comments
repo: 
```

File: _config.yml (hunk 1 above)
Comment (fletchto99) [reply _config.yml]: I like this idea! I'll do something like that.

File: _config.yml (hunk 3 above)
Comment (fletchto99) [reply _config.yml]: Good call. I'll update the wording to reflect that.

**Comment by fletchto99** [comment]
Rebased onto dev & changed the base branch for the PR.

**Comment by fletchto99** [comment]
> I'm not sure to understand. What do you mean exactly? Are you talking about your installation or in general if a user switch from Gitment to Gitalk?

I'm talking about if users switch from `Gitment` to `Gitalk`. This is because the Gitment implementation uses the permalink as the identifier for the issue: https://github.com/fletchto99/hexo-theme-tranquilpeak/blob/fletchto99/add-git-talk/layout/_partial/script.ejs#L33

While I now use the post title for the issue: https://github.com/fletchto99/hexo-theme-tranquilpeak/blob/fletchto99/add-git-talk/layout/_partial/script.ejs#L67

There are problems with both implementations. The `Gitment` implementation often generates URLs which are too long (issue labels can be a max of 50 chars). A simple solution was to use the post title which is usually less than that. The only problem I foresee with using the post title is that if you ever change the title this can break the relation between the GitHub issue and the gitalk comments. Though this can be fixed by manually updating the issue title on GitHub.

**Comment by fletchto99** [comment]
Pending CI changes & final review, this should implement the majority of the `gitalk` features.

I've omitted:
- `title`: As modifying this would break the relationship between the post and GitHub issue
- `body`: As there's no easy way to modify this per post, the default body is good enough
- `number`: As this would break the relationship between the post and GitHub issue
- `id`: As this is set dynamically to create the relationship between GitHub issue and post
- `admins`: As I figure the blog owner also owns the GitHub issue, so I just use the `owner` field for this.

Let me know if you would like these options added in some manner.

**Comment by LouisBarranqueiro** [comment]
tested and I was confused to have the latest comment at the top. This is really weird that this is the default option. Let's add `pagerDirection: "first"` in `theme.gitalk.options` to change this behavior.

**Comment by LouisBarranqueiro** [comment]
code and tests :lgtm:

**Comment by LouisBarranqueiro** [comment]
Thanks again @fletchto99 for the contribution. I really appreciate that! 👍

**Comment by fletchto99** [comment]
No problem! Happy to help give back even though my front end skills are meh at best :)

**Pull request merged**